Source/Scene/TimeDynamicImagery.js

```diff
@@ -0,0 +1,292 @@
+        };
+        if (!defined(approachingInterval) || !addToCache(this, tile, approachingInterval)) {
+            // Add to recent request list if we aren't approaching and interval or the request was throttled
+            tilesRequestedForInterval.push(tile);
+        }
+
+        // Don't let the tile list get out of hand
+        if (tilesRequestedForInterval.length >= 512) {
+            tilesRequestedForInterval.splice(0, 256);
+        }
+    };
+
+    TimeDynamicImagery.prototype._clockOnTick = function(clock) {
+        var time = clock.currentTime;
+        var times = this._times;
+        var index = times.indexOf(time);
+        var currentIntervalIndex = this._currentIntervalIndex;
+
+        if (index !== currentIntervalIndex) {
+            // Cancel all outstanding requests and clear out caches not from current time interval
+            var currentCache = this._tileCache[currentIntervalIndex];
+            for (var t in currentCache) {
+                if (currentCache.hasOwnProperty(t)) {
+                    currentCache[t].request.cancel();
+                }
+            }
+            delete this._tileCache[currentIntervalIndex];
+            this._tilesRequestedForInterval = [];
+
+            this._currentIntervalIndex = index;
+            this._reloadFunction();
+
+            return;
+        }
+
+        var approachingInterval = getApproachingInterval(this);
+        if (defined(approachingInterval)) {
+            // Start loading recent tiles from end of this._tilesRequestedForInterval
+            //  We keep preloading until we hit a throttling limit.
+            var tilesRequested = this._tilesRequestedForInterval;
+            var success = true;
+            while (success) {
+                if (tilesRequested.length === 0) {
+                    break;
+                }
+
+                var tile = tilesRequested.pop();
+                success = addToCache(this, tile, approachingInterval);
+                if (!success) {
+                    tilesRequested.push(tile);
+                }
+            }
+        }
+    };
+
+    function getKey(x, y, level) {
+        return x + '-' + y + '-' + level;
+    }
+
+    function getKeyElements(key) {
+        var s = key.split('-');
+        if (s.length !== 3) {
+            return undefined;
+        }
+
+        return {
+            x : Number(s[0]),
+            y : Number(s[1]),
+            level : Number(s[2])
+        };
+    }
+
+    function getApproachingInterval(that) {
+        var times = that._times;
+        if (!defined(times)) {
+            return undefined;
+        }
+        var clock = that._clock;
+        var time = clock.currentTime;
+        var isAnimating = clock.canAnimate && clock.shouldAnimate;
+        var multiplier = clock.multiplier;
+
+        if (!isAnimating && multiplier !== 0) {
+            return undefined;
+        }
+
+        var seconds;
+        var index = times.indexOf(time);
```

Comment:
Is it worth taking advantage of temporal coherence here? The index of the current time is more likely to be in the current , next or previous interval. Could save the linear search.